backend/pkg/api/connect/service/clusterstatus/redpanda.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package clusterstatus
+
+import (
+	"strings"
+
+	"github.com/redpanda-data/common-go/rpadmin"
+
+	consolev1alpha1 "github.com/redpanda-data/console/backend/pkg/protogen/redpanda/api/console/v1alpha1"
+)
+
+type redpandaStatusChecker struct{}
+
+// clusterVersionFromBrokerList returns the version of the Redpanda cluster. Since each broker
+// reports the version individually, we iterate through the list of brokers and
+// return the first reported version that contains a semVer.
+func (*redpandaStatusChecker) clusterVersionFromBrokerList(brokers []rpadmin.Broker) string {
+	//nolint:goconst // Other occurrences are unrelated
+	version := "unknown"
+	for _, broker := range brokers {
+		if broker.Version != "" {
+			// Broker version may look like this: "v22.1.4 - 491e56900d2316fcbb22aa1d37e7195897878309"
+			brokerVersion := strings.Split(broker.Version, " ")
+			if len(brokerVersion) > 0 {
+				version = "Redpanda " + brokerVersion[0]
+				break
+			}
+		}
+	}
+	return version
+}
+
+// ToPartitionBalancerViolationsProto maps a PartitionBalancerViolations Go type to its proto representation.
+func (*redpandaStatusChecker) partitionBalancerViolationsToProto(in rpadmin.PartitionBalancerViolations) *consolev1alpha1.GetRedpandaPartitionBalancerStatusResponse_Violations {
```

Comment:
Minor nit, comment does not match function name. Seems like originally at some point in the implementation this was public and got changed to private.